File: samples/scoping-samples/simple-stack-example-scoping-java/src/main/java/com/zhuinden/simplestackexamplescoping/features/words/NewWordKey.java
package com.zhuinden.simplestackexamplescoping.features.words;

import com.google.auto.value.AutoValue;
import com.zhuinden.simplestackexamplescoping.core.navigation.BaseKey;

import javax.annotation.Nonnull;

import androidx.fragment.app.Fragment;

@AutoValue
public abstract class NewWordKey
        extends BaseKey {
    public static NewWordKey create() {
        return new AutoValue_NewWordKey();
    }

    @Nonnull
    @Override
    protected Fragment instantiateFragment() {
        return new NewWordFragment();
    }
}

File: samples/scoping-samples/simple-stack-example-scoping/src/main/java/com/zhuinden/simplestackexamplescoping/features/words/NewWordKey.kt
package com.zhuinden.simplestackexamplescoping.features.words

import androidx.fragment.app.Fragment
import com.zhuinden.simplestack.ServiceBinder
import com.zhuinden.simplestackexamplescoping.core.navigation.BaseKey
import kotlinx.parcelize.Parcelize

/**
 * Created by Zhuinden on 2018.09.17.
 */
@Parcelize
data class NewWordKey(val placeholder: String = "") : BaseKey() {
    override fun instantiateFragment(): Fragment = NewWordFragment()

    override fun bindServices(serviceBinder: ServiceBinder) {}
}